src/core/storage/fileio/s3_api.cpp

```diff
@@ -136,8 +176,9 @@ const std::vector<std::string> S3Operation::_enum_to_str = {
  *
  * Returns true on success, false on failure.
  */
-bool parse_s3url(std::string url, s3url& ret, std::string& err_msg) {
+bool parse_s3url(std::string s3_url, s3url& ret, std::string& err_msg) {
```

Comment:
Let's do const std::string& out of habit here.

Reply:
inherited from previous impl. will do.

Reply:
Fair.  It wasn't the best code to begin with.